W-21433121 agent network 2.0 content ie (#660)
mulesoft/docs-code-builder: agent-network/2.0/modules/ROOT/pages/af-deploy-agent-network-targets.adoc

[[deploy-instances]]
= Deploy Agent Network Instances
:page-aliases: anypoint-code-builder::af-deploy-agent-network-targets.adoc

Deploy your agent network instance to a deployment target. You can deploy to a CloudHub 2.0 shared space, a CloudHub 2.0 private space, or to a Runtime Fabric. When you deploy, Omni Gateway secures inbound traffic to your agent network and outbound traffic from your brokers, agents, and MCP servers.

> These Omni Gateway policies are applied automatically at deployment.

Reviewer: ```suggestion
Omni Gateway applies these policies automatically at deployment.
```

[%header,cols="20,40a"]
|===
|Agent Type|Policy
|Broker
|

** A2A Agent Card policy
** Tracing policy
** Agent Connection Telemetry policy

|Agent

(ingress and egress)
|

** A2A Agent Card policy
** Tracing policy
** Agent Connection Telemetry policy

|MCP Server

(ingress and egress)
|

** MCP Support policy
** Tracing policy
** Agent Connection Telemetry policy

|===

[[before-you-begin]]
== Before You Begin

Make sure you review the xref:af-get-started.adoc#before-you-begin[prerequisites].

> When you deploy your agent network, an instance is deployed for each connection defined in your agent network. If your agent network contains brokers, an instance is deployed per broker.

Reviewer: ```suggestion
When you deploy your agent network, the system deploys one instance for each connection defined in your agent network. If your agent network contains brokers, the system deploys one instance per broker.
```

[[deploy-dev-agent]]
== Deploy Your Network Using MuleSoft Vibes

> MuleSoft Vibes can help you deploy your agent network instances. For more information, see xref:anypoint-code-builder::api-ai-create-spec.adoc[].

Reviewer: ```suggestion
Use MuleSoft Vibes to deploy your agent network instances. For more information, see xref:anypoint-code-builder::api-ai-create-spec.adoc[].
```

To get started, try one of these suggested messages.

* Deploy my agent network project to the production environment.
* Deploy the hello-world-network project to staging.
* Deploy agent network project at /path/to/project to dev environment with specific gateways.
* Deploy my agent network to CloudHub 2.0.

[[deploy-acb]]
== Deploy Your Network Using Anypoint Code Builder

Deploy your agent network instance to a CloudHub 2.0 space or Runtime Fabric target directly from Anypoint Code Builder.

. If you're not logged in already, log in to your Anypoint Platform account.
. In Anypoint Code Builder, choose one of the following:
  . In Explorer, right-click a project file and select *Deploy Agent Network*.
  . From the Command Palette, run this command: *MuleSoft: Deploy Agent Network*.
. In Deploy Agent Network, specify the environment, target space, gateways, and other deployment information. If you don't see any deployment targets to select, then you need to set up your target deployment space and set up gateways. For instructions, see xref:af-get-started.adoc[Get Started with Agent Networks].
. Select *Deploy*.

> . Your instances are deployed and ready to be tested. An instance is deployed for each connection defined in your agent network. If your agent network contains brokers, one instance is deployed per broker.

Reviewer: ```suggestion
. Your instances are now deployed and ready to test. The system deploys one instance for each connection defined in your agent network. If your agent network contains brokers, the system deploys one instance per broker.
```

Next, discover your agent network assets in *Agents & Tools* in Exchange, view your agent network in Agent Visualizer, apply a governance policy in API Manager, or monitor your agent network activity in Anypoint Monitoring.

[[deploy-cli]]
== Deploy a Network Using the Anypoint CLI

> If you run operations within a CI/CD environment, you can use Anypoint CLI's plugin to set up, create, build, publish, and deploy agent networks. For more information, see xref:af-build-agent-networks-in-a-ci-cd-environment.adoc[].

Reviewer: ```suggestion
If you run operations within a CI/CD environment, use Anypoint CLI's plugin to set up, create, build, publish, and deploy agent networks. For more information, see xref:af-build-agent-networks-in-a-ci-cd-environment.adoc[].
```

> If you run operations within a CI/CD environment, you can use Anypoint CLI's plugin to set up, create, build, publish, and deploy agent networks. For more information, see xref:af-build-agent-networks-in-a-ci-cd-environment.adoc[].

Reviewer: Add a See Also section?
